frontend/packages/ux-editor/src/components/config/editModal/EditOptions/EditCodeList/CodeListTableEditor.tsx

```diff
@@ -0,0 +1,74 @@
+import React, { useEffect, useState } from 'react';
+import type { IGenericEditComponent } from '@altinn/ux-editor/components/config/componentConfig';
+import type { SelectionComponentType } from '@altinn/ux-editor/types/FormComponent';
+import type { CodeListItem } from '@studio/components';
+import type { Option } from 'app-shared/types/Option';
+import { useTranslation } from 'react-i18next';
+import { StudioCodeListEditor, StudioModal } from '@studio/components';
+import { TableIcon } from '@studio/icons';
+import { useStudioEnvironmentParams } from 'app-shared/hooks/useStudioEnvironmentParams';
+import { useUpdateOptionListMutation } from '../../../../../hooks/mutations/useUpdateOptionListMutation';
+import { useOptionListsQuery } from '../../../../../hooks/queries/useOptionListsQuery';
+import {
+  convertCodeListItemListToOptionsList,
+  convertOptionsListToCodeListItemList,
+} from './utils/conversionUtils';
+import { useCodeListEditorTexts } from './hooks/useCodeListEditorTexts';
+import { usePreviewContext } from 'app-development/contexts/PreviewContext';
+import classes from './CodeListTableEditor.module.css';
+
+type CodeListTableEditorProps = Pick<IGenericEditComponent<SelectionComponentType>, 'component'>;
+
+export function CodeListTableEditor({ component }: CodeListTableEditorProps): React.ReactNode {
+  const { t } = useTranslation();
+  const { org, app } = useStudioEnvironmentParams();
+  const { doReloadPreview } = usePreviewContext();
+  const { data: optionsListMap, isFetching } = useOptionListsQuery(org, app);
+  const { mutate: uploadOptionList } = useUpdateOptionListMutation(org, app, {
+    hideDefaultError: true,
+  });
+  const [codeListItemList, setCodeListItemList] = useState<CodeListItem[]>([]);
+  const editorTexts = useCodeListEditorTexts();
+
+  useEffect(() => {
+    if (isFetching) return;
+    handleOptionsChange(optionsListMap[component.optionsId]);
+  }, [optionsListMap, component.optionsId, isFetching]);
```

Comment:
Does this need to be in a useEffect? Can we either use the `status` returned from the `useOptionListsQuery` and return a spinner, errorMessage or the content based on the respectively statuses; `loading`, `error` and `success`? 🤔

Reply:
I'll need the useEffect for when the user changes the optionsId, but i'll move the isFetching as it does not make sense here.

Reply:
Hm. But if the user changes what optionList that is connected to the component, will not this component update itself if we are sending a new component, with the updated connected optionListId, object to the handleComponentChange? Again, might be that I misunderstand the usage here